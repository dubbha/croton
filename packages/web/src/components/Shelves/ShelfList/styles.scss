.shelf-list {
  margin: auto;
  display: flex;
  flex-direction: row;
  align-items: stretch;
  align-content: stretch;
  justify-content: space-around;
  min-width: 70%;
}
