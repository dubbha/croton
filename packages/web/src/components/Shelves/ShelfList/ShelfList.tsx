import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { push } from 'connected-react-router';
import { getShelf } from 'store/shelf/selectors';
import { SHELF_GET_SHELVES } from 'store/shelf';
import { ShelfItem } from '../ShelfItem';
import { AddShelf } from '../AddShelf';

import './styles.scss';

export const ShelfList = () => {
  const dispatch = useDispatch();
  const { shelves } = useSelector(getShelf);

  useEffect(() => {
    dispatch({ type: SHELF_GET_SHELVES });
  }, [dispatch]);

  const selectShelf = (id: number) => dispatch(push(`/profile/shelf/${id}`));
  const refreshShelves = () => dispatch({ type: SHELF_GET_SHELVES });

  return (
    <div className="shelf-list row">
      {shelves.map((shelf) => (
        <ShelfItem
          key={shelf.id}
          id={shelf.id}
          name={shelf.name}
          location={shelf.location}
          description={shelf.description}
          onSelect={selectShelf}
        />
      ))}
      <AddShelf onClose={refreshShelves} />
    </div>
  );
};
