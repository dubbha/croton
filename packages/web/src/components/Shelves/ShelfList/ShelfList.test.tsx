import React from 'react';
import { render } from '@testing-library/react';
import { ShelfList } from './ShelfList';

jest.mock('connected-react-router', () => ({
  push: (path: string) => ({ type: 'callHistoryMethod', payload: { path } }),
}));

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: () => ({ shelves: [{ id: 'id', name: 'name', location: 'description' }] }),
}));

describe('components/ShelfList', () => {
  it('should render successfully', () => {
    const { container } = render(<ShelfList />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
