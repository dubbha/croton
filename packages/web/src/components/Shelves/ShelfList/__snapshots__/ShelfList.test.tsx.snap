// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ShelfList should render successfully 1`] = `
<div
  class="shelf-list row"
>
  <div
    class="shelf-item"
  >
    <div
      class="card text-primary"
    >
      <img
        class="card-img-top"
        src=""
      />
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          name
        </div>
        <div
          class="mb-2 text-muted card-subtitle h6"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-map-marker-alt fa-w-12 "
            data-icon="map-marker-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
              fill="currentColor"
            />
          </svg>
          <span
            class="icon-prefix"
          >
            location
          </span>
        </div>
        <p
          class="card-text"
        >
          description
        </p>
      </div>
    </div>
  </div>
  <div
    class="shelf-item"
  >
    <div
      class="card text-primary"
    >
      <div
        class="shelf-item-new"
      >
        <div
          class="card-title h5"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-plus fa-w-14 "
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
            />
          </svg>
          <span
            class="icon-prefix"
          >
            Add New Shelf
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
